internal/provider/resource_host_catalog_plugin.go

```diff
@@ -0,0 +1,413 @@
+	"github.com/hashicorp/boundary/api/hostcatalogs"
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	hostCatalogTypePlugin = "plugin"
+)
+
+func resourceHostCatalogPlugin() *schema.Resource {
+	return &schema.Resource{
+		Description: "The host catalog resource allows you to configure a Boundary plugin-type host catalog. Host " +
+			"catalogs are always part of a project, so a project resource should be used inline or you " +
+			"should have the project ID in hand to successfully configure a host catalog.",
+
+		CreateContext: resourceHostCatalogPluginCreate,
+		ReadContext:   resourceHostCatalogPluginRead,
+		UpdateContext: resourceHostCatalogPluginUpdate,
+		DeleteContext: resourceHostCatalogPluginDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the host catalog.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description: "The host catalog name. Defaults to the resource name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			DescriptionKey: {
+				Description: "The host catalog description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			ScopeIdKey: {
+				Description: "The scope ID in which the resource is created.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			PluginIdKey: {
+				Description: "The ID of the plugin that should back the resource. This or " + PluginNameKey + " must be defined.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true, // If name is provided this will be computed
+			},
+			PluginNameKey: {
+				Description: "The name of the plugin that should back the resource. This or " + PluginIdKey + " must be defined.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			TypeKey: {
+				Description: "The host catalog type. Only `plugin` is supported, and is the default.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Default:     "plugin",
+			},
+			AttributesJsonKey: {
+				Description: `The attributes for the host catalog. Either values encoded with the "jsonencode" function, pre-escaped JSON string, or a file:// or env:// path. Set to a string "null" or remove the block to clear all attributes in the host catalog.`,
+				Type:        schema.TypeString,
+				Optional:    true,
+				// If set to null in config and nothing comes from API, consider
+				// it the same. Same if config changes from empty to null.
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					switch {
+					case old == new:
```

Comment:
Why would go be doing the encoding?  The "new" string will be whatever the end user provides into the field wont it?  Or is this diffing newly read data from the API against what was read previously?  I thought it was the former since SchemaDiffSuppressFunc compares diffs in the Schema field (as opposed to the ResourceData).

Reply:
Resolving this as with the recent mini-RFC there is now an agreed upon way forward.

Reply:
The mini rfc, if it is the one I am thinking of, discusses secrets.  This is the attributes field and is type string.  The input could be anything, even json that the user tries to hand craft themselves.  This is why I think we want to normalize the value entered here before doing the diff otherwise we may always see a diff in that situation.

Reply:
Why do you think that's not the behavior we want?

If the user changes the input, let's diff it. Do you think we should do something else?

Reply:
Addressed!